persistence/src/commonMain/db_user/com/wire/kalium/persistence/MessageAssetView.sq

```diff
@@ -81,3 +81,13 @@ AND assetMimeType NOT IN :mimeTypes
 AND assetId IS NOT NULL
 AND expireAfterMillis IS NULL
 ORDER BY date DESC;
+
+getAllAssetMessagesByConversationId:
+SELECT assetId FROM MessageAssetView
+WHERE conversationId = :conversationId
+AND visibility IN :visibility
+AND contentType IN :contentTypes
+AND assetId IS NOT NULL
+AND isEphemeral = FALSE
+AND dataPath IS NOT NULL
+ORDER BY date DESC;
```

Comment:
if you want to query for assets to delete there is no need to filter by visibility or isEphemeral jsut get all and start the delete process, also since there is a lot to delete maybe it is worth it to account for interruptions 
1. mark the conversation that it need to be cleared 
2. start a service that does the deletion
3. delete once all data have been deleted

this way it will not block sync the whole time this is running

Reply:
Going to remove the `visibility` and `isEphemeral`. And yes for the handling of the use case its a matter of the `client` to either run it in a service or display a progress until the deletion is done - in our case it will most likely be an service